// @strapi/admin/dist/_chunks/constants-4aca55b6.js
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const schema = require("./schema-260fec28.js");
const constants = require("./constants-d5d67ca8.js");
require("react");
require("react-dnd");
require("@reduxjs/toolkit");
require("./index-4de70b5d.js");
require("react/jsx-runtime");
require("@strapi/helper-plugin");
require("react-dom/client");
require("@strapi/design-system");
require("invariant");
require("lodash/isFunction");
require("lodash/merge");
require("lodash/pick");
require("react-helmet");
require("react-router-dom");
require("react-intl");
require("react-redux");
require("@radix-ui/react-context");
require("@strapi/design-system/v2");
require("@strapi/icons");
require("formik");
require("lodash/camelCase");
require("react-query");
require("styled-components");
require("yup");
require("lodash/defaultsDeep");
require("lodash/omit");
require("qs");
require("immer");
require("lodash/get");
require("lodash/set");
require("lodash/isBoolean");
require("lodash/isEmpty");
require("lodash/isNaN");
require("lodash/toNumber");
const REVIEW_WORKFLOW_COLUMNS_EE = [
  {
    key: `__${constants.STAGE_ATTRIBUTE_NAME}_temp_key__`,
    name: constants.STAGE_ATTRIBUTE_NAME,
    fieldSchema: {
      type: "relation"
    },
    metadatas: {
      // formatMessage() will be applied when the column is rendered
      label: {
        id: schema.getTrad(`containers.ListPage.table-headers.reviewWorkflows.stage`),
        defaultMessage: "Review stage"
      },
      searchable: false,
      sortable: true,
      mainField: {
        name: "name",
        schema: {
          type: "string"
        }
      }
    }
  },
  {
    key: `__${constants.ASSIGNEE_ATTRIBUTE_NAME}_temp_key__`,
    name: constants.ASSIGNEE_ATTRIBUTE_NAME,
    fieldSchema: {
      type: "relation"
    },
    metadatas: {
      label: {
        id: schema.getTrad(`containers.ListPage.table-headers.reviewWorkflows.assignee`),
        defaultMessage: "Assignee"
      },
      searchable: false,
      sortable: true,
      mainField: {
        name: "firstname",
        schema: {
          type: "string"
        }
      }
    }
  }
];
exports.REVIEW_WORKFLOW_COLUMNS_EE = REVIEW_WORKFLOW_COLUMNS_EE;
