// @strapi/admin/dist/_chunks/index-176cc121.mjs
import { jsxs, Fragment, jsx } from "react/jsx-runtime";
import { Button, HeaderLayout, Flex, IconButton, Select, Option, Typography, Textarea, TextInput } from "@strapi/design-system";
import { useNotification, useFetchClient, useAPIErrorHandler, ConfirmDialog, Link, useTracking, useClipboard, ContentBox, translatedErrors } from "@strapi/helper-plugin";
import { Refresh, Check, ArrowLeft, Duplicate, Key } from "@strapi/icons";
import PropTypes from "prop-types";
import { useIntl } from "react-intl";
import { useState } from "react";
import { AxiosError } from "axios";
import { format, addDays } from "date-fns";
import * as locales from "date-fns/locale";
import * as yup from "yup";
const useRegenerate = (url, id, onRegenerate, onError) => {
  const [isLoadingConfirmation, setIsLoadingConfirmation] = useState(false);
  const toggleNotification = useNotification();
  const { post } = useFetchClient();
  const { formatAPIError } = useAPIErrorHandler();
  const regenerateData = async () => {
    try {
      const {
        data: {
          data: { accessKey }
        }
      } = await post(`${url}${id}/regenerate`);
      setIsLoadingConfirmation(false);
      onRegenerate(accessKey);
    } catch (error) {
      setIsLoadingConfirmation(false);
      if (onError) {
        onError(error);
      } else {
        if (error instanceof AxiosError) {
          toggleNotification({
            type: "warning",
            message: formatAPIError(error)
          });
        }
      }
    }
  };
  return {
    regenerateData,
    isLoadingConfirmation
  };
};
const Regenerate = ({ onRegenerate, idToRegenerate, backUrl, onError }) => {
  const { formatMessage } = useIntl();
  const [showConfirmDialog, setShowConfirmDialog] = useState(false);
  const { regenerateData, isLoadingConfirmation } = useRegenerate(
    backUrl,
    idToRegenerate,
    onRegenerate,
    onError
  );
  const handleConfirmRegeneration = async () => {
    regenerateData();
    setShowConfirmDialog(false);
  };
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(
      Button,
      {
        startIcon: /* @__PURE__ */ jsx(Refresh, {}),
        type: "button",
        size: "S",
        variant: "tertiary",
        onClick: () => setShowConfirmDialog(true),
        name: "regenerate",
        children: formatMessage({
          id: "Settings.tokens.regenerate",
          defaultMessage: "Regenerate"
        })
      }
    ),
    /* @__PURE__ */ jsx(
      ConfirmDialog,
      {
        bodyText: {
          id: "Settings.tokens.popUpWarning.message",
          defaultMessage: "Are you sure you want to regenerate this token?"
        },
        iconRightButton: /* @__PURE__ */ jsx(Refresh, {}),
        isConfirmButtonLoading: isLoadingConfirmation,
        isOpen: showConfirmDialog,
        onToggleDialog: () => setShowConfirmDialog(false),
        onConfirm: handleConfirmRegeneration,
        leftButtonText: {
          id: "Settings.tokens.Button.cancel",
          defaultMessage: "Cancel"
        },
        rightButtonText: {
          id: "Settings.tokens.Button.regenerate",
          defaultMessage: "Regenerate"
        },
        title: {
          id: "Settings.tokens.RegenerateDialog.title",
          defaultMessage: "Regenerate token"
        }
      }
    )
  ] });
};
Regenerate.defaultProps = { onRegenerate() {
}, onError: void 0 };
Regenerate.propTypes = {
  onRegenerate: PropTypes.func,
  idToRegenerate: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  backUrl: PropTypes.string.isRequired,
  onError: PropTypes.func
};
const FormHead = ({
  title,
  token,
  setToken,
  canEditInputs,
  canRegenerate,
  isSubmitting,
  backUrl,
  regenerateUrl,
  onErrorRegenerate
}) => {
  const { formatMessage } = useIntl();
  const handleRegenerate = (newKey) => {
    setToken({
      ...token,
      accessKey: newKey
    });
  };
  return /* @__PURE__ */ jsx(
    HeaderLayout,
    {
      title: token?.name || formatMessage(title),
      primaryAction: canEditInputs ? /* @__PURE__ */ jsxs(Flex, { gap: 2, children: [
        canRegenerate && token?.id && /* @__PURE__ */ jsx(
          Regenerate,
          {
            backUrl: regenerateUrl,
            onRegenerate: handleRegenerate,
            idToRegenerate: token?.id,
            onError: onErrorRegenerate
          }
        ),
        /* @__PURE__ */ jsx(
          Button,
          {
            disabled: isSubmitting,
            loading: isSubmitting,
            startIcon: /* @__PURE__ */ jsx(Check, {}),
            type: "submit",
            size: "S",
            children: formatMessage({
              id: "global.save",
              defaultMessage: "Save"
            })
          }
        )
      ] }) : canRegenerate && token?.id && /* @__PURE__ */ jsx(
        Regenerate,
        {
          onRegenerate: handleRegenerate,
          idToRegenerate: token?.id,
          backUrl: regenerateUrl
        }
      ),
      navigationAction: /* @__PURE__ */ jsx(Link, { startIcon: /* @__PURE__ */ jsx(ArrowLeft, {}), to: backUrl, children: formatMessage({
        id: "global.back",
        defaultMessage: "Back"
      }) }),
      ellipsis: true
    }
  );
};
FormHead.propTypes = {
  token: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    type: PropTypes.string,
    lifespan: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    name: PropTypes.string,
    accessKey: PropTypes.string,
    permissions: PropTypes.array,
    description: PropTypes.string,
    createdAt: PropTypes.string
  }),
  canEditInputs: PropTypes.bool.isRequired,
  canRegenerate: PropTypes.bool.isRequired,
  setToken: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  backUrl: PropTypes.string.isRequired,
  title: PropTypes.shape({
    id: PropTypes.string,
    label: PropTypes.string
  }).isRequired,
  regenerateUrl: PropTypes.string.isRequired,
  onErrorRegenerate: PropTypes.func
};
FormHead.defaultProps = {
  token: void 0,
  onErrorRegenerate: void 0
};
const TokenBox = ({ token, tokenType }) => {
  const { formatMessage } = useIntl();
  const toggleNotification = useNotification();
  const { trackUsage } = useTracking();
  const { copy } = useClipboard();
  const handleClick = (token2) => async () => {
    const didCopy = await copy(token2);
    if (didCopy) {
      trackUsage("didCopyTokenKey", {
        tokenType
      });
      toggleNotification({
        type: "success",
        message: { id: "Settings.tokens.notification.copied" }
      });
    }
  };
  return /* @__PURE__ */ jsx(
    ContentBox,
    {
      endAction: token && /* @__PURE__ */ jsx("span", { style: { alignSelf: "start" }, children: /* @__PURE__ */ jsx(
        IconButton,
        {
          label: formatMessage({
            id: "app.component.CopyToClipboard.label",
            defaultMessage: "Copy to clipboard"
          }),
          onClick: handleClick(token),
          noBorder: true,
          icon: /* @__PURE__ */ jsx(Duplicate, {}),
          style: { padding: 0, height: "1rem" }
        }
      ) }),
      title: token || formatMessage({
        id: "Settings.tokens.copy.editTitle",
        defaultMessage: "This token isn’t accessible anymore."
      }),
      subtitle: token ? formatMessage({
        id: "Settings.tokens.copy.lastWarning",
        defaultMessage: "Make sure to copy this token, you won’t be able to see it again!"
      }) : formatMessage({
        id: "Settings.tokens.copy.editMessage",
        defaultMessage: "For security reasons, you can only see your token once."
      }),
      icon: /* @__PURE__ */ jsx(Key, {}),
      iconBackground: "neutral100"
    }
  );
};
TokenBox.defaultProps = {
  token: null
};
TokenBox.propTypes = {
  token: PropTypes.string,
  tokenType: PropTypes.string.isRequired
};
const getDateOfExpiration = (createdAt, duration, language = "en") => {
  if (duration && typeof duration === "number") {
    const durationInDays = duration / 24 / 60 / 60 / 1e3;
    return format(addDays(new Date(createdAt), durationInDays), "PPP", {
      locale: locales[language]
    });
  }
  return "Unlimited";
};
const schema = yup.object().shape({
  name: yup.string(translatedErrors.string).max(100).required(translatedErrors.required),
  type: yup.string(translatedErrors.string).oneOf(["read-only", "full-access", "custom"]).required(translatedErrors.required),
  description: yup.string().nullable(),
  lifespan: yup.number().integer().min(0).nullable().defined(translatedErrors.required)
});
const LifeSpanInput = ({ token, errors, values, onChange, isCreating }) => {
  const { formatMessage } = useIntl();
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsxs(
      Select,
      {
        name: "lifespan",
        label: formatMessage({
          id: "Settings.tokens.form.duration",
          defaultMessage: "Token duration"
        }),
        value: values.lifespan !== null ? values.lifespan : "0",
        error: errors.lifespan ? formatMessage(
          errors.lifespan?.id ? errors.lifespan : { id: errors.lifespan, defaultMessage: errors.lifespan }
        ) : null,
        onChange: (value) => {
          onChange({ target: { name: "lifespan", value } });
        },
        required: true,
        disabled: !isCreating,
        placeholder: "Select",
        children: [
          /* @__PURE__ */ jsx(Option, { value: "604800000", children: formatMessage({
            id: "Settings.tokens.duration.7-days",
            defaultMessage: "7 days"
          }) }),
          /* @__PURE__ */ jsx(Option, { value: "2592000000", children: formatMessage({
            id: "Settings.tokens.duration.30-days",
            defaultMessage: "30 days"
          }) }),
          /* @__PURE__ */ jsx(Option, { value: "7776000000", children: formatMessage({
            id: "Settings.tokens.duration.90-days",
            defaultMessage: "90 days"
          }) }),
          /* @__PURE__ */ jsx(Option, { value: "0", children: formatMessage({
            id: "Settings.tokens.duration.unlimited",
            defaultMessage: "Unlimited"
          }) })
        ]
      }
    ),
    /* @__PURE__ */ jsx(Typography, { variant: "pi", textColor: "neutral600", children: !isCreating && `${formatMessage({
      id: "Settings.tokens.duration.expiration-date",
      defaultMessage: "Expiration date"
    })}: ${getDateOfExpiration(token?.createdAt, parseInt(values.lifespan, 10))}` })
  ] });
};
LifeSpanInput.propTypes = {
  errors: PropTypes.shape({
    lifespan: PropTypes.string
  }),
  onChange: PropTypes.func.isRequired,
  values: PropTypes.shape({
    lifespan: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
  }).isRequired,
  isCreating: PropTypes.bool.isRequired,
  token: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    type: PropTypes.string,
    lifespan: PropTypes.string,
    name: PropTypes.string,
    accessKey: PropTypes.string,
    permissions: PropTypes.array,
    description: PropTypes.string,
    createdAt: PropTypes.string
  })
};
LifeSpanInput.defaultProps = {
  errors: {},
  token: {}
};
const TokenDescription = ({ errors, values, onChange, canEditInputs }) => {
  const { formatMessage } = useIntl();
  return /* @__PURE__ */ jsx(
    Textarea,
    {
      label: formatMessage({
        id: "Settings.tokens.form.description",
        defaultMessage: "Description"
      }),
      id: "description",
      error: errors.description ? formatMessage(
        errors.description?.id ? errors.description : {
          id: errors.description,
          defaultMessage: errors.description
        }
      ) : null,
      onChange,
      disabled: !canEditInputs,
      children: values.description
    }
  );
};
TokenDescription.propTypes = {
  errors: PropTypes.shape({
    description: PropTypes.string
  }),
  onChange: PropTypes.func.isRequired,
  canEditInputs: PropTypes.bool.isRequired,
  values: PropTypes.shape({
    description: PropTypes.string
  }).isRequired
};
TokenDescription.defaultProps = {
  errors: {}
};
const TokenName = ({ errors, values, onChange, canEditInputs }) => {
  const { formatMessage } = useIntl();
  return /* @__PURE__ */ jsx(
    TextInput,
    {
      name: "name",
      error: errors.name ? formatMessage(
        errors.name?.id ? errors.name : { id: errors.name, defaultMessage: errors.name }
      ) : null,
      label: formatMessage({
        id: "Settings.tokens.form.name",
        defaultMessage: "Name"
      }),
      onChange,
      value: values.name,
      disabled: !canEditInputs,
      required: true
    }
  );
};
TokenName.propTypes = {
  errors: PropTypes.shape({
    name: PropTypes.string
  }),
  onChange: PropTypes.func.isRequired,
  canEditInputs: PropTypes.bool.isRequired,
  values: PropTypes.shape({
    name: PropTypes.string
  }).isRequired
};
TokenName.defaultProps = {
  errors: {}
};
const TokenTypeSelect = ({ name, errors, values, onChange, canEditInputs, options, label }) => {
  const { formatMessage } = useIntl();
  return /* @__PURE__ */ jsx(
    Select,
    {
      name,
      label: formatMessage({
        id: label.id,
        defaultMessage: label.defaultMessage
      }),
      value: values && values[name],
      error: errors[name] ? formatMessage(
        errors[name]?.id ? errors[name] : { id: errors[name], defaultMessage: errors[name] }
      ) : null,
      onChange,
      placeholder: "Select",
      required: true,
      disabled: !canEditInputs,
      children: options && options.map(({ value, label: label2 }) => /* @__PURE__ */ jsx(Option, { value, children: formatMessage(label2) }, value))
    }
  );
};
TokenTypeSelect.propTypes = {
  name: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.shape({
        id: PropTypes.string,
        defaultMessage: PropTypes.string
      }),
      value: PropTypes.string
    })
  ),
  errors: PropTypes.shape({
    type: PropTypes.string
  }),
  onChange: PropTypes.func.isRequired,
  canEditInputs: PropTypes.bool.isRequired,
  values: PropTypes.shape({
    type: PropTypes.string
  }).isRequired,
  label: PropTypes.shape({
    id: PropTypes.string,
    defaultMessage: PropTypes.string
  }).isRequired
};
TokenTypeSelect.defaultProps = {
  name: "type",
  errors: {},
  options: []
};
export {
  FormHead as F,
  LifeSpanInput as L,
  TokenName as T,
  TokenDescription as a,
  TokenTypeSelect as b,
  TokenBox as c,
  schema as s
};
