// @strapi/admin/dist/_chunks/index-cb6a72d2.mjs
import { jsxs, jsx } from "react/jsx-runtime";
import { Flex, Box, Typography } from "@strapi/design-system";
import { pxToRem } from "@strapi/helper-plugin";
import PropTypes from "prop-types";
import { b as STAGE_COLOR_DEFAULT } from "./constants-8092eeb5.mjs";
import { g as getStageColorByHex } from "./colors-e7489dd4.mjs";
import { useIntl } from "react-intl";
import "lodash/cloneDeep";
import "lodash/get";
import "lodash/omit";
import "lodash/isNaN";
import "./schema-6bd8bd85.mjs";
import "lodash/merge";
import "lodash/set";
import { g as getDisplayName } from "./index-bd18f803.mjs";
import "react";
import "react-dnd";
import "@reduxjs/toolkit";
import "./index-8933483b.mjs";
import "react-dom/client";
import "invariant";
import "lodash/isFunction";
import "lodash/pick";
import "react-helmet";
import "react-router-dom";
import "react-redux";
import "@radix-ui/react-context";
import "@strapi/design-system/v2";
import "@strapi/icons";
import "formik";
import "lodash/camelCase";
import "react-query";
import "styled-components";
import "yup";
import "lodash/defaultsDeep";
import "qs";
import "immer";
import "lodash/isBoolean";
import "lodash/isEmpty";
import "lodash/toNumber";
import "lodash/sortBy";
import "./DragLayer-bff96835.mjs";
import "./AuthenticatedApp-bb31f306.mjs";
import "semver/functions/lt";
import "semver/functions/valid";
import "react-dnd-html5-backend";
import "react-error-boundary";
import "axios";
import "lodash/isObject";
import "lodash/isEqual";
import "lodash/upperFirst";
import "lodash/size";
import "lodash/take";
import "slate";
import "slate-history";
import "slate-react";
import "@radix-ui/react-toolbar";
import "./useDebounce-9364f340.mjs";
import "react-window";
import "codemirror5";
import "highlight.js";
import "markdown-it";
import "markdown-it-abbr";
import "markdown-it-container";
import "markdown-it-deflist";
import "markdown-it-emoji";
import "markdown-it-footnote";
import "markdown-it-ins";
import "markdown-it-mark";
import "markdown-it-sub";
import "markdown-it-sup";
import "sanitize-html";
import "codemirror5/addon/display/placeholder";
import "lodash/toLower";
import "lodash/toString";
import "react-dom";
import "fractional-indexing";
import "lodash/uniqBy";
import "lodash/unset";
import "lodash/isArray";
import "lodash/flatMap";
import "lodash/uniq";
import "./useAdminUsers-09f86a58.mjs";
import "./formatAPIErrors-db6ed675.mjs";
import "date-fns/parseISO";
import "lodash/isNumber";
import "lodash/groupBy";
function ReviewWorkflowsStageEE({ color, name }) {
  const { themeColorName } = getStageColorByHex(color);
  return /* @__PURE__ */ jsxs(Flex, { alignItems: "center", gap: 2, maxWidth: pxToRem(300), children: [
    /* @__PURE__ */ jsx(
      Box,
      {
        height: 2,
        background: color,
        borderColor: themeColorName === "neutral0" ? "neutral150" : "transparent",
        hasRadius: true,
        shrink: 0,
        width: 2
      }
    ),
    /* @__PURE__ */ jsx(Typography, { fontWeight: "regular", textColor: "neutral700", ellipsis: true, children: name })
  ] });
}
ReviewWorkflowsStageEE.defaultProps = {
  color: STAGE_COLOR_DEFAULT
};
ReviewWorkflowsStageEE.propTypes = {
  color: PropTypes.string,
  name: PropTypes.string.isRequired
};
function ReviewWorkflowsAssigneeEE({ user }) {
  const { formatMessage } = useIntl();
  return /* @__PURE__ */ jsx(Typography, { textColor: "neutral800", children: getDisplayName(user, formatMessage) });
}
ReviewWorkflowsAssigneeEE.propTypes = {
  user: PropTypes.shape({
    firstname: PropTypes.string,
    lastname: PropTypes.string,
    username: PropTypes.string
  }).isRequired
};
export {
  ReviewWorkflowsAssigneeEE,
  ReviewWorkflowsStageEE
};
