// @strapi/admin/dist/_chunks/index-fc700e52.js
"use strict";
const jsxRuntime = require("react/jsx-runtime");
const designSystem = require("@strapi/design-system");
const helperPlugin = require("@strapi/helper-plugin");
const PropTypes = require("prop-types");
const reactIntl = require("react-intl");
const reactRouterDom = require("react-router-dom");
const React = require("react");
const Icons = require("@strapi/icons");
const styled = require("styled-components");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const PropTypes__default = /* @__PURE__ */ _interopDefault(PropTypes);
const styled__default = /* @__PURE__ */ _interopDefault(styled);
const DeleteButton = ({ tokenName, onClickDelete, tokenType }) => {
  const { formatMessage } = reactIntl.useIntl();
  const { trackUsage } = helperPlugin.useTracking();
  const [showConfirmDialog, setShowConfirmDialog] = React.useState(false);
  const handleClickDelete = () => {
    setShowConfirmDialog(false);
    trackUsage("willDeleteToken", {
      tokenType
    });
    onClickDelete();
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(designSystem.Box, { paddingLeft: 1, onClick: (e) => e.stopPropagation(), children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      designSystem.IconButton,
      {
        onClick: () => {
          setShowConfirmDialog(true);
        },
        label: formatMessage(
          {
            id: "global.delete-target",
            defaultMessage: "Delete {target}"
          },
          { target: `${tokenName}` }
        ),
        name: "delete",
        noBorder: true,
        icon: /* @__PURE__ */ jsxRuntime.jsx(Icons.Trash, {})
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx(
      helperPlugin.ConfirmDialog,
      {
        onToggleDialog: () => setShowConfirmDialog(false),
        onConfirm: handleClickDelete,
        isOpen: showConfirmDialog
      }
    )
  ] });
};
DeleteButton.propTypes = {
  tokenName: PropTypes__default.default.string.isRequired,
  onClickDelete: PropTypes__default.default.func.isRequired,
  tokenType: PropTypes__default.default.string.isRequired
};
const MESSAGES_MAP = {
  edit: {
    id: "app.component.table.edit",
    defaultMessage: "Edit {target}"
  },
  read: {
    id: "app.component.table.read",
    defaultMessage: "Read {target}"
  }
};
const LinkStyled = styled__default.default(helperPlugin.Link)`
  svg {
    path {
      fill: ${({ theme }) => theme.colors.neutral500};
    }
  }

  &:hover,
  &:focus {
    svg {
      path {
        fill: ${({ theme }) => theme.colors.neutral800};
      }
    }
  }
`;
const DefaultButton = ({ tokenName, tokenId, buttonType, children }) => {
  const { formatMessage } = reactIntl.useIntl();
  const {
    location: { pathname }
  } = reactRouterDom.useHistory();
  return /* @__PURE__ */ jsxRuntime.jsx(
    LinkStyled,
    {
      to: `${pathname}/${tokenId}`,
      title: formatMessage(MESSAGES_MAP[buttonType], { target: tokenName }),
      children
    }
  );
};
DefaultButton.propTypes = {
  tokenName: PropTypes__default.default.string.isRequired,
  tokenId: PropTypes__default.default.oneOfType([PropTypes__default.default.string, PropTypes__default.default.number]).isRequired,
  buttonType: PropTypes__default.default.string,
  children: PropTypes__default.default.node.isRequired
};
DefaultButton.defaultProps = {
  buttonType: "edit"
};
const ReadButton = ({ tokenName, tokenId }) => {
  return /* @__PURE__ */ jsxRuntime.jsx(DefaultButton, { tokenName, tokenId, buttonType: "read", children: /* @__PURE__ */ jsxRuntime.jsx(Icons.Eye, {}) });
};
ReadButton.propTypes = {
  tokenName: PropTypes__default.default.string.isRequired,
  tokenId: PropTypes__default.default.oneOfType([PropTypes__default.default.string, PropTypes__default.default.number]).isRequired
};
const UpdateButton = ({ tokenName, tokenId }) => {
  return /* @__PURE__ */ jsxRuntime.jsx(DefaultButton, { tokenName, tokenId, children: /* @__PURE__ */ jsxRuntime.jsx(Icons.Pencil, {}) });
};
UpdateButton.propTypes = {
  tokenName: PropTypes__default.default.string.isRequired,
  tokenId: PropTypes__default.default.oneOfType([PropTypes__default.default.string, PropTypes__default.default.number]).isRequired
};
const Table = ({
  permissions,
  headers,
  contentType,
  isLoading,
  tokens,
  onConfirmDelete,
  tokenType
}) => {
  const { canDelete, canUpdate, canRead } = permissions;
  return /* @__PURE__ */ jsxRuntime.jsx(
    helperPlugin.DynamicTable,
    {
      headers,
      contentType,
      rows: tokens,
      withBulkActions: canDelete || canUpdate || canRead,
      isLoading,
      onConfirmDelete,
      children: /* @__PURE__ */ jsxRuntime.jsx(
        TableRows,
        {
          tokenType,
          permissions,
          onConfirmDelete
        }
      )
    }
  );
};
Table.propTypes = {
  tokens: PropTypes__default.default.array,
  permissions: PropTypes__default.default.shape({
    canRead: PropTypes__default.default.bool,
    canDelete: PropTypes__default.default.bool,
    canUpdate: PropTypes__default.default.bool
  }).isRequired,
  headers: PropTypes__default.default.arrayOf(
    PropTypes__default.default.shape({
      cellFormatter: PropTypes__default.default.func,
      key: PropTypes__default.default.string.isRequired,
      metadatas: PropTypes__default.default.shape({
        label: PropTypes__default.default.string.isRequired,
        sortable: PropTypes__default.default.bool
      }).isRequired,
      name: PropTypes__default.default.string.isRequired
    })
  ),
  contentType: PropTypes__default.default.string.isRequired,
  isLoading: PropTypes__default.default.bool,
  onConfirmDelete: PropTypes__default.default.func,
  tokenType: PropTypes__default.default.string.isRequired
};
Table.defaultProps = {
  tokens: [],
  headers: [],
  isLoading: false,
  onConfirmDelete() {
  }
};
const TableRows = ({ tokenType, permissions, rows, withBulkActions, onConfirmDelete }) => {
  const { canDelete, canUpdate, canRead } = permissions;
  const [{ query }] = helperPlugin.useQueryParams();
  const { formatMessage } = reactIntl.useIntl();
  const [, sortOrder] = query ? query.sort.split(":") : "ASC";
  const {
    push,
    location: { pathname }
  } = reactRouterDom.useHistory();
  const { trackUsage } = helperPlugin.useTracking();
  const sortedTokens = rows.sort((a, b) => {
    const comparison = a.name.localeCompare(b.name);
    return sortOrder === "DESC" ? -comparison : comparison;
  });
  return /* @__PURE__ */ jsxRuntime.jsx(designSystem.Tbody, { children: sortedTokens.map((token) => {
    return /* @__PURE__ */ jsxRuntime.jsxs(
      designSystem.Tr,
      {
        ...helperPlugin.onRowClick({
          fn() {
            trackUsage("willEditTokenFromList", {
              tokenType
            });
            push(`${pathname}/${token.id}`);
          },
          condition: canUpdate
        }),
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(designSystem.Td, { maxWidth: helperPlugin.pxToRem(250), children: /* @__PURE__ */ jsxRuntime.jsx(designSystem.Typography, { textColor: "neutral800", fontWeight: "bold", ellipsis: true, children: token.name }) }),
          /* @__PURE__ */ jsxRuntime.jsx(designSystem.Td, { maxWidth: helperPlugin.pxToRem(250), children: /* @__PURE__ */ jsxRuntime.jsx(designSystem.Typography, { textColor: "neutral800", ellipsis: true, children: token.description }) }),
          /* @__PURE__ */ jsxRuntime.jsx(designSystem.Td, { children: /* @__PURE__ */ jsxRuntime.jsx(designSystem.Typography, { textColor: "neutral800", children: /* @__PURE__ */ jsxRuntime.jsx(helperPlugin.RelativeTime, { timestamp: new Date(token.createdAt) }) }) }),
          /* @__PURE__ */ jsxRuntime.jsx(designSystem.Td, { children: token.lastUsedAt && /* @__PURE__ */ jsxRuntime.jsx(designSystem.Typography, { textColor: "neutral800", children: /* @__PURE__ */ jsxRuntime.jsx(
            helperPlugin.RelativeTime,
            {
              timestamp: new Date(token.lastUsedAt),
              customIntervals: [
                {
                  unit: "hours",
                  threshold: 1,
                  text: formatMessage({
                    id: "Settings.apiTokens.lastHour",
                    defaultMessage: "last hour"
                  })
                }
              ]
            }
          ) }) }),
          withBulkActions && /* @__PURE__ */ jsxRuntime.jsx(designSystem.Td, { children: /* @__PURE__ */ jsxRuntime.jsxs(designSystem.Flex, { justifyContent: "end", children: [
            canUpdate && /* @__PURE__ */ jsxRuntime.jsx(UpdateButton, { tokenName: token.name, tokenId: token.id }),
            !canUpdate && canRead && /* @__PURE__ */ jsxRuntime.jsx(ReadButton, { tokenName: token.name, tokenId: token.id }),
            canDelete && /* @__PURE__ */ jsxRuntime.jsx(
              DeleteButton,
              {
                tokenName: token.name,
                onClickDelete: () => onConfirmDelete(token.id),
                tokenType
              }
            )
          ] }) })
        ]
      },
      token.id
    );
  }) });
};
TableRows.defaultProps = {
  rows: [],
  withBulkActions: false
};
TableRows.propTypes = {
  rows: PropTypes__default.default.array,
  permissions: PropTypes__default.default.shape({
    canRead: PropTypes__default.default.bool,
    canDelete: PropTypes__default.default.bool,
    canUpdate: PropTypes__default.default.bool
  }).isRequired,
  onConfirmDelete: PropTypes__default.default.func.isRequired,
  tokenType: PropTypes__default.default.string.isRequired,
  withBulkActions: PropTypes__default.default.bool
};
exports.Table = Table;
