// @strapi/admin/dist/_chunks/index-88298941.js
"use strict";
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const designSystem = require("@strapi/design-system");
const helperPlugin = require("@strapi/helper-plugin");
const Icons = require("@strapi/icons");
const PropTypes = require("prop-types");
const reactIntl = require("react-intl");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const PropTypes__default = /* @__PURE__ */ _interopDefault(PropTypes);
const Filters = ({ displayedFilters }) => {
  const [isVisible, setIsVisible] = React.useState(false);
  const { formatMessage } = reactIntl.useIntl();
  const buttonRef = React.useRef();
  const handleToggle = () => {
    setIsVisible((prev) => !prev);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(designSystem.Box, { paddingTop: 1, paddingBottom: 1, children: [
      /* @__PURE__ */ jsxRuntime.jsx(
        designSystem.Button,
        {
          variant: "tertiary",
          ref: buttonRef,
          startIcon: /* @__PURE__ */ jsxRuntime.jsx(Icons.Filter, {}),
          onClick: handleToggle,
          size: "S",
          children: formatMessage({ id: "app.utils.filters", defaultMessage: "Filters" })
        }
      ),
      isVisible && /* @__PURE__ */ jsxRuntime.jsx(
        helperPlugin.FilterPopoverURLQuery,
        {
          displayedFilters,
          isVisible,
          onToggle: handleToggle,
          source: buttonRef
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntime.jsx(helperPlugin.FilterListURLQuery, { filtersSchema: displayedFilters })
  ] });
};
Filters.propTypes = {
  displayedFilters: PropTypes__default.default.arrayOf(
    PropTypes__default.default.shape({
      name: PropTypes__default.default.string.isRequired,
      metadatas: PropTypes__default.default.shape({ label: PropTypes__default.default.string }),
      fieldSchema: PropTypes__default.default.shape({ type: PropTypes__default.default.string })
    })
  ).isRequired
};
exports.Filters = Filters;
